server/src/utils.ts

```
@@ -135,3 +149,56 @@ export function parseBooksXML(booksXmlPath: string): {
     publicRoot: bookVars['PUBLIC_ROOT'] ?? '/interactives',
   };
 }
+
+export function mergeByIndex(
+  lhs: unknown[],
+  rhs: unknown[],
+  options?: MergeOptions,
+) {
+  return Object.values(
+    recursiveMerge(
+      Object.fromEntries(Object.entries(lhs)),
+      Object.fromEntries(Object.entries(rhs)),
+      options,
+    ) as object,
+  );
+}
+
+export function defaultValueMerge(lhs: unknown, rhs: unknown) {
+  return rhs ?? lhs;
+}
+
+export function recursiveMerge(
```

:sunglasses:

One thing though... what if recursiveMerge errored if both the left side and right side were non-empty? I think that case silently succeeds right?

That would catch cases like "Why isn't the answer showing up? I put the answer in the (public) JSON but it still doesn't show up (because it's also defined in the private JSON)"